crates/matrix-sdk-ui/tests/integration/main.rs

```diff
@@ -105,3 +108,77 @@ async fn mock_encryption_state(server: &MockServer, is_encrypted: bool) {
             .await;
     }
 }
+
+#[derive(Copy, Clone)]
+struct SlidingSyncMatcher;
+
+impl Match for SlidingSyncMatcher {
+    fn matches(&self, request: &Request) -> bool {
+        request.url.path() == "/_matrix/client/unstable/org.matrix.msc3575/sync"
+            && request.method == Method::Post
+    }
+}
+
+#[macro_export]
+macro_rules! sync_then_assert_request_and_fake_response {
```

Comment:
This macro is really tailored for SS. Is it the correct place to get it, esp. with this name? I don't think so :-).